user/pages/03.commerce2/02.developer-guide/03.core/01.conditions/docs.md

```diff
@@ -4,8 +4,284 @@ taxonomy:
+Methods** for an order.
+
+
+Example user stories / use cases:
+---
+
+  **Payment Gateway:** As a store manager I will only accept credit card 
+  payments from customers with US billing addresses. 
+  Use: [OrderBillingAddress - Billing address](https://cgit.drupalcode.org/commerce/tree/modules/order/src/Plugin/Commerce/Condition/OrderBillingAddress.php)
+  
+  **Shipping:** As a store manager I want to offer flat rate shipping for 
+  only certain products. 
+  Use: [OrderProduct - Order contains specific products](https://cgit.drupalcode.org/commerce/tree/modules/product/src/Plugin/Commerce/Condition/OrderProduct.php)
+
+  **Promotion:** As a store manager I want to encourage bulk ordering by 
+  creating a promotion that applies to orders of at least 5 products. 
+  Use: [OrderItemQuantity - Total discounted product quantity](https://cgit.drupalcode.org/commerce/tree/modules/promotion/src/Plugin/Commerce/Condition/OrderItemQuantity.php)
+  This use case will be used as an example for the rest of this tutorial. 
+  Note: This use case will only be fully realized when the new promotion is
+  given a coupon and the coupon is applied to an order during checkout.  
+  
+  ![Screenshot of add Promotion UI](promotion-example.png)
+  
+
+Minimum requirements to build a condition:
+---
+
+As you can see from comparing 
+[ConditionInterface](https://cgit.drupalcode.org/commerce/tree/src/Plugin/Commerce/Condition/ConditionInterface.php)
+with [ConditionBase](https://cgit.drupalcode.org/commerce/tree/src/Plugin/Commerce/Condition/ConditionBase.php),
+there is only one function required when extending ConditionBase to fulfill 
+the ConditionInterface definition of ***evaluate***. 
+
+```
+interface ConditionInterface extends ConfigurablePluginInterface, PluginFormInterface, PluginInspectionInterface {
+  /** * Gets the condition label. */ 
+  public function getLabel();
+  /** * Shown in the condition UI when enabling/disabling a condition. */
+  public function getDisplayLabel();
+  /** * This is the entity type ID of the entity passed to evaluate(). */
+  public function getEntityTypeId();
+
+  /**
+   * Evaluates the condition.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity.
+   *
+   * @return bool
+   *   TRUE if the condition has been met, FALSE otherwise.
+   */
+  public function evaluate(EntityInterface $entity);
+}
+
+```
+
+Functionally, a condition might not be useful without some sort of user 
+configuration. `defaultConfiguration`, `buildConfigurationForm`, and 
+`validateConfigurationForm` go hand in hand with `evaluate` to create a basic 
+condition.
+
+Lets see how these functions are used in practice by analysing one of 
+the simplest conditions to ship with commerce:  [OrderItemQuantity](https://cgit.drupalcode.org/commerce/tree/modules/promotion/src/Plugin/Commerce/Condition/OrderItemQuantity.php)
+
+
+Annotations
+---
+
+First, meta-information about the condition must be declared in the 
```

Comment:
Describe what annotations are and link to Drupal docs about them:

https://www.drupal.org/docs/8/api/plugin-api/annotations-based-plugins